test/Yesod/Paginator/WidgetsSpec.hs

```diff
@@ -189,3 +195,48 @@ spec = withApp $ do
             , "<li class=\"next disabled\"><a>»</a></li>"
             , "</ul>"
             ]
+
+data Params = Params
+    { paramsPageParamName :: PageParamName
+    , paramsPageNumber :: Int
+    , paramsParams :: [(Text, Text)]
+    }
+    deriving Show
+
+instance Arbitrary Params where
+    arbitrary = do
+        params <- listOf $ liftArbitrary2 genText genText
+        pageNumber <- getPositive <$> arbitrary
+        pageParamName <- PageParamName <$> genText
+        pure $ Params pageParamName pageNumber params
+      where
+        genText :: Gen Text
+        genText = listOf (choose ('a', 'z')) <&> pack
```

Comment:
Maybe there's a better generator for parameters (considered e.g. [this](https://hackage.haskell.org/package/quickcheck-text-0.1.2.1/docs/Test-QuickCheck-Utf8.html#v:genValidUtf8)), but settled for this simple approach for now.